.circleci/config/commands/go_test.yml

```
@@ -58,21 +58,38 @@ steps:
         USE_DOCKER=1
         <</ parameters.use_docker >>
 
-        # Split Go tests by prior test times.  If use_docker is true, only run
-        # tests that depend on docker, otherwise only those that don't.
-        if [ $USE_DOCKER == 1 ]; then
-          package_names=$(go list -test -json ./... |
-            jq -r 'select(.Deps != null) |
-              select(any(.Deps[] ; contains("github.com/hashicorp/vault/helper/testhelpers/docker"))) |
-              .ForTest | select(. != null)' |
-              sort -u | grep -v vault/integ | circleci tests split --split-by=timings --timings-type=classname)
-        else
-          package_names=$(go list -test -json ./... |
-            jq -r 'select(.Deps != null) |
-              select(all(.Deps[] ; contains("github.com/hashicorp/vault/helper/testhelpers/docker")|not)) |
-              .ForTest | select(. != null)' |
-              sort -u | grep -v vault/integ | circleci tests split --split-by=timings --timings-type=classname)
-        fi
+        # Check all directories with a go.mod file
+        modules=('.' 'api' 'sdk')
+        all_package_names=""
+
+        for dir in "${modules[@]}"
+        do
+          cd $dir
```

I'm not sure if `cd .` works, but it might be worth double-checking.

---

Ah yeah, I did check this, it does work i.e it no-ops and doesn't move anywhere

---

```suggestion
          cd "$dir" || exit
```

See https://github.com/koalaman/shellcheck/wiki/SC2103 for more details.

---

bf0407d